src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -54,4 +109,79 @@ String constructFirecloudGroupName(String duosId) {
   String constructUniqueFirecloudGroupName(String duosId) {
     return String.format("%s-%s", constructFirecloudGroupName(duosId), UUID.randomUUID());
   }
+
+  /**
+   * If the DUOS dataset ID is registered in TDR, force a sync of its Firecloud group members.
+   *
+   * @param duosId DUOS dataset ID
+   * @return DUOS Firecloud group reflecting the last successful sync of its members
+   */
+  public DuosFirecloudGroupModel syncDuosDatasetAuthorizedUsers(String duosId) {
+    DuosFirecloudGroupModel firecloudGroup = retrieveFirecloudGroup(duosId);
+    return syncFirecloudGroupContents(firecloudGroup);
+  }
+
+  /**
+   * @param firecloudGroup DUOS Firecloud group whose members will be overwritten by the latest set
+   *     of approved users
+   * @return DUOS Firecloud group reflecting the last successful sync of its members
+   */
+  @VisibleForTesting
+  DuosFirecloudGroupModel syncFirecloudGroupContents(DuosFirecloudGroupModel firecloudGroup) {
+    Instant lastSyncedDate = Instant.now();
+    iamService.overwriteGroupPolicyEmails(
+        firecloudGroup.getFirecloudGroupName(),
+        IamRole.MEMBER.toString(),
+        getAuthorizedUsers(firecloudGroup.getDuosId()));
+    updateLastSynced(firecloudGroup, lastSyncedDate);
+    return duosDao.retrieveFirecloudGroup(firecloudGroup.getId());
+  }
+
+  /**
+   * @param firecloudGroup DUOS Firecloud group whose members were overwritten by the latest set of
+   *     approved users
+   * @param lastSyncedDate DUOS Firecloud group members match the DUOS dataset's authorized users at
+   *     this moment in time
+   */
+  @VisibleForTesting
+  void updateLastSynced(DuosFirecloudGroupModel firecloudGroup, Instant lastSyncedDate) {
+    UUID id = firecloudGroup.getId();
+    try {
+      duosDao.updateFirecloudGroupLastSyncedDate(id, lastSyncedDate);
+    } catch (CannotSerializeTransactionException ex) {
+      String msg =
+          String.join(
+              " ",
+              "Members of Firecloud group",
+              firecloudGroup.getFirecloudGroupName(),
+              "were updated, but could not update DUOS Firecloud Group record",
+              id.toString(),
+              "with last_synced_date",
+              lastSyncedDate.toString());
```

Comment:
This use of `String.join()` seems unusual to me. Normally `join()` is called on a collection or list of variables. In this case it looks like another way of writing a string concatenation. Using `+` has the benefit that you don't need to all `toString()`, and (to me) seems like a more idiomatic way to write this in Java. Another way to write this would be to use `formatted()`:
```suggestion
      String msg =
          ("Members of Firecloud group %s were updated, but could not update DUOS Firecloud " +
              "Group record %s with last_synced_date %s")
              .formatted(firecloudGroup.getFirecloudGroupName(), id, lastSyncedDate);
```